CONTRIBUTING.md

```diff
@@ -6,6 +6,23 @@ See the [Kubeaddons Contributing Documentation](https://github.com/mesosphere/ku
 
 Additionally see the sections below for notes about other rules and considerations for contributions.
 
+## Addon Delete Upgrade Strategy
+
+It's possible to add an annotation to an Addon which will trigger a "delete style upgrade" (literally, removal and reinstallation) for the addon under certain version conditions, e.g.:
+
+```yaml
+apiVersion: kubeaddons.mesosphere.io/v1beta2
+kind: ClusterAddon
+metadata:
+  name: dashboard
+    # versions of the dashboard older than v2 are not directly compatible and so a delete uprade is needed in this case to avoid conflicts with the older resources.
+    helm.kubeaddons.mesosphere.io/upgrade-strategy: "[{\"upgradeFrom\": \"<=2.0.0\", \"strategy\": \"delete\"}]"
+    # to maintain compatibility with older versions of the kubeaddons controller, please duplicate this with the old key:
```

Comment:
This sucks, we should be checking this internally not having the user duplicate this... no happy.

Reply:
I agree, but I feel like since right now we're just trying to improve documentation a little, this isn't the right scope?